src/components/advanced-color-picker/advanced-color-picker.component.js

```diff
@@ -50,8 +50,20 @@ const AdvancedColorPicker = ({
     }
   }, [colors, currentColor, dialogOpen, isOpen]);
 
-  const handleFocus = useCallback(() => {
-    selectedColorRef.current.input.current.focus();
+  const handleFocus = useCallback((e, firstFocusableElement, lastFocusableElement) => {
+    if (e.key === 'Tab' || e.keyCode === 9) {
+      /* istanbul ignore else */
+      if (e.shiftKey) {
+        /* istanbul ignore else */
+        if (document.activeElement === selectedColorRef.current.input.current) {
+          lastFocusableElement.focus();
+          e.preventDefault();
+        }
+      } else if (document.activeElement === lastFocusableElement) {
+        selectedColorRef.current.input.current.focus();
```

Comment:
This won't work after this https://github.com/Sage/carbon/pull/3031 PR get's merged

Reply:
No problem, I'll wait until that's merged and sort it out

Reply:
Hopefully my PR gets resolved soon. If not then we'll let this one go, and I'll solve this issue on my branch then, just wanted to leave a note to not forget about it later on :)

Reply:
Note we have discussed and are happy for this PR to be merged first